library/std/src/sys/unix/process/process_unix.rs

```diff
@@ -267,7 +345,6 @@ impl Command {
     #[cfg(any(
         target_os = "macos",
         target_os = "freebsd",
-        all(target_os = "linux", target_env = "gnu")
```

Comment:
Could we continue using posix_spawn on GNU/Linux, while adding pidfd to a list of exceptions? It should end up using CLONE_VFORK & CLONE_VM, and looking at various benchmarks it can make substantial performance difference for some use-cases.

Reply:
The clone3 implementation could use CLONE_VFORK to achieve the same.
For reference, Go seems to pull it off while only having to [jump though one hoop to deal with the shared memory](https://github.com/golang/go/commit/67e537541c043c701001f002bed0cda70ce72767).

Reply:
Using `posix_spawn` would require us to create the pidfd in the child process, and then use a Unix domain socket to transfer it to the parent. Using `clone3` with the proper flags seems much simpler.

Reply:
When pidfd is not used, so there should be no impact on the implementation. See #77455 for an additional motivation.